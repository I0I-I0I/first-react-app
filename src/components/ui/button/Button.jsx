import styles from "./Button.module.css";

const classes = {
	size: {
		small: `${styles.buttonSmall}`,
		large: `${styles.buttonLarge}`,
	},
};

const Button = ({ children, size = "small", ...props }) => (
	<button className={`${styles.button} ${classes.size[size]}`} {...props}>
		{children}
	</button>
);

export default Button;
