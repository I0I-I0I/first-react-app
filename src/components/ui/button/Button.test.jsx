import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Button from "./Button.jsx";

describe("Button", () => {
	it("correct display passed text", () => {
		render(<Button>Test</Button>);
		expect(screen.getByText("Test")).toBeTruthy();
	});

	it("get correct props", () => {
		render(
			<Button size="large" disabled>
				Test
			</Button>,
		);

		const button = screen.getByRole("button");

		expect(button.disabled).toBeTruthy();
		expect(button.className).toMatch(/buttonLarge/);
	});

	it("snapshot", () => {
		const { container } = render(
			<Button size="large" disabled>
				Test
			</Button>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
