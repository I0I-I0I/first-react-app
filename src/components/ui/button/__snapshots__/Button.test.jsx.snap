// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > snapshot 1`] = `
<button
  class="_button_3c5126 _buttonLarge_3c5126"
  disabled=""
>
  Test
</button>
`;
